From: Lorenzo Stoakes <lorenzo.stoakes@oracle.com>
Subject: mm: fix incorrect mixedmap implementation
Date: Mon, 15 Sep 2025 11:01:06 +0100

This was typo'd due to staring too long at the cramfs implementation.

Link: https://lkml.kernel.org/r/4ce3adda-6351-4530-92aa-103acf638004@lucifer.local
Signed-off-by: Lorenzo Stoakes <lorenzo.stoakes@oracle.com>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---

 mm/util.c |    5 ++---
 1 file changed, 2 insertions(+), 3 deletions(-)

--- a/mm/util.c~mm-add-ability-to-take-further-action-in-vm_area_desc-fix-2
+++ a/mm/util.c
@@ -1364,15 +1364,14 @@ int mmap_action_complete(struct mmap_act
 		unsigned long pgnum = 0;
 		unsigned long pfn = action->mixedmap.pfn;
 		unsigned long addr = action->mixedmap.addr;
-		unsigned long vaddr = vma->vm_start;
 
 		VM_WARN_ON_ONCE(!(vma->vm_flags & VM_MIXEDMAP));
 
 		for (; pgnum < action->mixedmap.num_pages;
-		     pgnum++, pfn++, addr += PAGE_SIZE, vaddr += PAGE_SIZE) {
+		    pgnum++, pfn++, addr += PAGE_SIZE) {
 			vm_fault_t vmf;
 
-			vmf = vmf_insert_mixed(vma, vaddr, addr);
+			vmf = vmf_insert_mixed(vma, addr, pfn);
 			if (vmf & VM_FAULT_ERROR) {
 				err = vm_fault_to_errno(vmf, 0);
 				break;
_
